R/MsBackendMongoDB.R

```diff
@@ -194,46 +199,44 @@ setMethod(
   function(object, dbcon, data, ...) {
     if (missing(dbcon))
       stop("Parameter 'dbcon' is required for 'MsBackendMongoDb'")
-    
+
     # Insert data if provided
     if (!missing(data)) {
-      .createMsBackendMongoDb_new(dbcon, x = data)
+      .createMsBackendMongoDb(dbcon, x = data)
     }
-    
+
     # Validate MongoDb connection
     msg <- .valid_mongocon(dbcon)
     if (!is.null(msg)) stop(msg)
-    
+
     # Store collections names
     object@.collections <- dbcon
-    
+
     # Fetch spectrum_id_ as character
     spectrum_ids <- as.character(
       dbcon[["ms_spectrum_coll"]]$distinct("spectrum_id_")
     )
-    
+
     # Assign spectrum IDs and nspectra
     object@spectraIds <- spectrum_ids
     object@id_map <- spectrum_ids
     object@nspectra <- length(spectrum_ids)
     object@dbcon <- dbcon
     object@peak_fun <- .fetch_peaks_data_long_mongo
-    
+
     # Ensure localData has correct number of rows
     object@localData <- data.frame(dummy = rep(NA_integer_, object@nspectra))
-    
+
+    # Get available metadata column names from the ms_spectrum_coll
+    svars <- ""
```

Comment:
please get here the spectrum metadata column names (column names of the ms_spectrum_coll collection)